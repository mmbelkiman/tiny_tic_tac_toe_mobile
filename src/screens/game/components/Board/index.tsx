import {View} from 'react-native';
import React from 'react';
import {PlayerMark} from '@game/common/types';
import styles from './styles';
import Square from './components/Square';

interface BoardProps {
  board: PlayerMark[][];
  nextPlayerIsCircle: boolean;
  disabled: boolean;
  squareSize: number;
  onClickSquare: (rowIndex: number, colIndex: number) => void;
  circleColor: string;
  crossColor: string;
}

const Board: React.FC<BoardProps> = ({
  board,
  nextPlayerIsCircle,
  squareSize,
  disabled,
  onClickSquare,
  crossColor,
  circleColor,
}) => {
  if (!board || board.length === 0) {
    return null;
  }

  return (
    <>
      {board.map((row, rowIndex) => (
        <View key={rowIndex} style={styles.row}>
          {row.map((square, colIndex) => (
            <Square
              key={colIndex}
              playerMarkIsCircle={nextPlayerIsCircle}
              squareSize={squareSize}
              disabled={disabled}
              crossColor={crossColor}
              circleColor={circleColor}
              onPress={() => onClickSquare(rowIndex, colIndex)}
            />
          ))}
        </View>
      ))}
    </>
  );
};

export default Board;
