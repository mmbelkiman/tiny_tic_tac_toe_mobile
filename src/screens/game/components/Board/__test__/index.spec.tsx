import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Board from '../index';
import {act} from 'react-test-renderer';

const sampleBoard = [
  ['X', 'O', 'X'],
  ['O', ' ', 'O'],
  ['X', 'O', 'X'],
];

describe('Board Component', () => {
  it('renders correctly with sample board data', () => {
    const mockOnClickSquare = jest.fn();

    const {getAllByTestId} = render(
      <Board
        board={sampleBoard}
        nextPlayerIsCircle={false}
        disabled={false}
        squareSize={50}
        onClickSquare={mockOnClickSquare}
        circleColor="red"
        crossColor="blue"
      />,
    );

    const squareElements: any = getAllByTestId(/^Square/);
    expect(squareElements.length).toBe(9);
  });

  it('calls onClickSquare when a square is pressed', async () => {
    const mockOnClickSquare = jest.fn();

    const {getByTestId} = render(
      <Board
        board={sampleBoard}
        nextPlayerIsCircle={false}
        disabled={false}
        squareSize={50}
        onClickSquare={mockOnClickSquare}
        circleColor="red"
        crossColor="blue"
      />,
    );

    const firstSquare = getByTestId('Square0.0.TouchableOpacity');
    fireEvent.press(firstSquare);

    return act(() => {
      expect(mockOnClickSquare).toHaveBeenCalledWith(0, 0);
    });
  });

  it('matches snapshot with sample board data', () => {
    const {toJSON} = render(
      <Board
        board={sampleBoard}
        nextPlayerIsCircle={false}
        disabled={false}
        squareSize={50}
        onClickSquare={jest.fn()}
        circleColor="red"
        crossColor="blue"
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with clear board', () => {
    const {toJSON} = render(
      <Board
        board={[]}
        nextPlayerIsCircle={false}
        disabled={false}
        squareSize={50}
        onClickSquare={jest.fn()}
        circleColor="red"
        crossColor="blue"
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
