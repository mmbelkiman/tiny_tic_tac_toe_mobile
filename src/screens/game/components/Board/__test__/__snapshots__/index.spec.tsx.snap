// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component matches snapshot with clear board 1`] = `null`;

exports[`Board Component matches snapshot with sample board data 1`] = `
[
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square0.0.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square0.1.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square0.2.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square1.0.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square1.1.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square1.2.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square2.0.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square2.1.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Square2.2.TouchableOpacity"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#e9e9e9",
            "borderColor": "#fff",
            "borderWidth": 2,
            "height": 50,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": -2,
              },
            ],
            "width": 50,
          }
        }
      />
    </View>
  </View>,
]
`;
